### PR TITLE
Small code refactoring

### PR DESCRIPTION
**Pull request by videodanchik**
This is a small refactoring before the tedlium3 `conformer_ctc2` recipe. Also, extend `1best` decoding with LM weights scales.

### CODE REVIEW

File: icefall/decode.py [hunk 1]
```diff
@@ -468,11 +469,28 @@ def one_best_decoding(
...
+        for lm_scale in lm_scale_list:
+            am_scores = saved_am_scores / lm_scale
+            lattice.scores = am_scores + lattice.lm_scores
+
+            best_path = k2.shortest_path(
+                lattice, use_double_scores=use_double_scores
+            )
+            key = f"lm_scale_{lm_scale}"
+            ans[key] = best_path
+        return ans
```
Comment (csukuangfj) [inline icefall/decode.py:491]: Could you also update the type hint of the return value of this function?

File: icefall/decode.py [hunk 2]
```diff
@@ -459,6 +459,7 @@ def build_levenshtein_graphs(self) -> k2.Fsa:
 def one_best_decoding(
     lattice: k2.Fsa,
     use_double_scores: bool = True,
+    lm_scale_list: Optional[List[float]] = None,
```
Comment (csukuangfj) [inline icefall/decode.py:462]: Is there an example decoding script that is using it?

We found today that it helps by tuning this value for HLG decoding.

Comment (videodanchik) [reply icefall/decode.py:462]: Yes I have an example it is in my current tedlium3 `conformer_ctc2` recipe it is almost ready (give me one or two days, and I will do PR). I also discovered during my early work with k2/icefall that tuning this parameter affects WER dramatically during HLG decoding. My current best model achieves SOTA on tedlium3 with HLG decoding and tuning language model weight
image

File: icefall/decode.py (hunk 1 above)
Comment (videodanchik) [reply icefall/decode.py:491]: sure

**Comment by videodanchik** [comment]
@csukuangfj I fixed the type hint.

**Comment by csukuangfj** [comment]
Thanks!

**Pull request merged**